w3d_translator/read_xml.py

```diff
@@ -50,51 +49,74 @@ def read_xml(file):
     return story
 
 
-def parse_camera(xml: ET.Element) -> dict:
-    return {
-        "far_clip": float(xml.attrib["far-clip"]),
-        "placement": parse_placement(xml.find("Placement")),
-    }
+def parse_string(string: str) -> Union[bool, int, float, tuple, str]:
+    # Check if string is a boolean
+    try:
+        test = string.lower()
+        if test == "true":
+            return True
+        elif test == "false":
+            return False
+    except AttributeError:
+        pass
 
+    # Check if string is an integer
+    try:
+        return int(string)
+    except ValueError:
+        pass
 
-def parse_placement(xml: ET.Element) -> dict:
-    relative_to = xml.find("RelativeTo").text
-    position = str_to_tuple(xml.find("Position").text)
-    axis = None
-    look_at = None
-
-    axis_xml = xml.find("Axis")
-    if axis_xml is not None:
-        axis = {
-            "rotation": str_to_tuple(axis_xml.attrib["rotation"]),
-            "angle": float(axis_xml.attrib["angle"]),
-        }
-
-    look_at_xml = xml.find("LookAt")
-    if look_at_xml is not None:
-        axis = {
-            "target": str_to_tuple(look_at_xml.attrib["target"]),
-            "up": str_to_tuple(look_at_xml.attrib["up"]),
-        }
+    # Check if string is a float
+    try:
+        return float(string)
+    except ValueError:
+        pass
 
-    return {
-        "relative_to": relative_to,
-        "position": position,
-        "Axis": axis,
-        "LookAt": look_at,
-    }
+    # Check if string is a tuple (of integers or floats)
+    test = string.replace("(", "").replace(")", "").split(",")
```

Comment:
a regex check seems probably a safer bet for this one

Reply:
Using regex now! Took a bit but I think I have it down